dom/src/HTMLSource.ts

```diff
@@ -1,35 +1,48 @@
 import xs, {Stream} from 'xstream';
-import {StreamAdapter, DevToolEnabledSource} from '@cycle/base';
-import {DOMSource, EventsFnOptions} from './DOMSource';
-import xsSA from '@cycle/xstream-adapter';
-
-export class HTMLSource implements DOMSource {
-  private _html$: any;
-  private _empty$: any;
-
-  constructor(html$: Stream<string>,
-              private runSA: StreamAdapter,
-              private _name: string) {
-    this._html$ = html$;
-    this._empty$ = runSA.adapt(xs.empty(), xsSA.streamSubscribe);
+import {DevToolEnabledSource} from '@cycle/base';
+import {DOMSource} from './DOMSource';
+import {DOMSourceOptions} from './DOMSourceOptions';
+import {EventsFnOptions} from './EventsFnOptions';
+import xsAdapter from '@cycle/xstream-adapter';
+
+export interface HTMLSourceOptions extends DOMSourceOptions {
+  html$: Stream<string>;
+}
+
+export class HTMLSource extends DOMSource {
+  private _html$: Stream<string>;
+  private _empty$: Stream<any>;
```

Comment:
I got it from this https://github.com/cyclejs/cyclejs/blob/master/dom/src/HTMLSource.ts#L10